docs/reference/command/readstate.adoc

```diff
@@ -13,3 +13,20 @@ if the result is not `null` and not an empty string, otherwise it invokes the `e
 ----
 
 NOTE: `read-state` will run the `then` commands even if `${{use_https}}` is `"false"` because that is not empty and not null. Use link:./js.adoc['js'] if you want to perform boolean operations
+
+If you need to evaluate two variables or more you can use one of the following approaches:
+
+* use `js` which lets you write a JavaScript function that invokes the `then` if the return is truth, otherwise it invokes the else
+* use `${{=` to start a JavaScript evaluation and perform the boolean operation there
+e.g.
+
+[source,yaml]
+----
+- read-state: ${{= ${{abort_on_failure:true}} && "${{START_JOB_RESULT:}}" !== 'SUCCESS' }}
+  then:
+  - sh: echo Failure
+  else:
+  - sh: echo Success
```

Comment:
Instead of using `read-state`, isn't better using `js` directly to evalute boolean expressions?
```
- js: ${{abort_on_failure:true}} && "${{START_JOB_RESULT:}}" !== 'SUCCESS' 
  then:
  - sh: echo Failure
  else:
  - sh: echo Success
```
I remember @willr3 mentioned this in the past, this I'd like to get his advice here 😄

Reply:
To get it working as you are proposing I think you are missing a `regex`, am I right?
```yaml
- read-state: ${{= ${{abort_on_failure:true}} && "${{START_JOB_RESULT:}}" !== 'SUCCESS' }}
- regex: true
  then:
  - sh: echo Failure
  else:
  - sh: echo Success
```

Reply:
both work, a boolean operation probably makes more sense as a `js` command because it means we no longer need the `regex` but I think having both examples is useful. Perhaps we steer users toward `js` but show that it can also be accomplished in `read-state`?

Reply:
We can document how to do it with JavaScript as well. WDYT?